plugin/pkg/shoot/validator/admission.go

```diff
@@ -385,9 +385,13 @@ func validateProvider(c *validationContext) field.ErrorList {
 		)...)
 	}
 
-	ok, validKubernetesVersions, versionDefault := validateKubernetesVersionConstraints(c.cloudProfile.Spec.Kubernetes.Versions, c.shoot.Spec.Kubernetes.Version, c.oldShoot.Spec.Kubernetes.Version)
+	ok, isDefaulted, validKubernetesVersions, versionDefault := validateKubernetesVersionConstraints(c.cloudProfile.Spec.Kubernetes.Versions, c.shoot.Spec.Kubernetes.Version, c.oldShoot.Spec.Kubernetes.Version)
 	if !ok {
-		allErrs = append(allErrs, field.NotSupported(field.NewPath("spec", "kubernetes", "version"), c.shoot.Spec.Kubernetes.Version, validKubernetesVersions))
+		err := field.NotSupported(field.NewPath("spec", "kubernetes", "version"), c.shoot.Spec.Kubernetes.Version, validKubernetesVersions)
+		if isDefaulted {
+			err.Detail = fmt.Sprintf("unable to default version. No higher non-preview version found. In case there is a higher preview version available for that minor, it has to be selected explicitly -  %s", err.Detail)
```

Comment:
What does "no higher non-preview version" mean? I'm confused by the word "higher" ?!

Reply:
meaning there is no version x > version to default.  
WDYT about "no greater non-preview version"

Reply:
Hm, maybe I'm missing something. We are defaulting the Kubernetes version when somebody is using `1.18`, for example. We try to find the highest patch version of `1.18` that is `supported`, right?
Hence, if we can't find such a version, wouldn't a message like "unable to default version - couldn't find a patch version for 1.18 with \"supported\" classification - in case there is a \"preview\"-classified version please select it explicitly" be more accurate?

Reply:
It only looks for the highest non-preview version. That can be a supported version or be deprecated.

Reply:
OK, understood. Then:

"unable to default version - couldn't find a suitable patch version for 1.18 (either \"supported\" or \"deprecated\" classification and a non-expired expiration date) - in case there is a \"preview\"-classified version please select it explicitly"

Reply:
Actually a version can also have no classification - can we leave the supported and deprecated out? 

"unable to default version - couldn't find a suitable patch version for 1.18. Suitable patch versions have a non-expired expiration date and are no "preview" versions. "Preview"-classified versions have to be selected explicitly"

Reply:
sounds good!

Reply:
PTAL